README.md

```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+The API is simple, it either returns a 200 status Code with the JSON body, or it returns a 4xx or 5xx status code, in which case the rest-client library throws an exception.
 
-Basically, the mapping is done this way:
+Now, what if we want to add a new cat by calling "POST http://api.catsforall.com/cats"? We do so like this:
 
-* If the HTTP response has a *4XX* or *5XX* status code, we return `EitherStrict.Left(error)`, where `error` indicates information about the error (it has the error message, whether the error was a 4XX or a 5XX one, etc).
-* If the HTTP response has a *401* status code, we return `EitherStrict.Right(OptionStrict.Nothing)`
-* If the HTTP response has a *2XX* status code, we return `EitherStrict.Right(OptionStrict.Just(operationResult))`, where `operationResult` is the data typed obtained by deserializing the JSON response body.
+![restclient example 1](docs/img/restclient_cat_example_post.png?raw=true)
 
-No matter what the API responds with, we return the user a type specifying the information he wants from the API, mapping the HTTP response to a C# type (that is composed of various types like `EitherStrict`, `OptionStrict`, etc).
+We can also do the usual PUT and DELETE calls:
 
-### Processors
+![restclient example 1](docs/img/restclient_cat_example_put.png?raw=true)
 
-To achieve this, rest-client uses a pipeline of processors that check whether they find a specific HTTP status code, and if they find it they return a value of the returning type.
+![restclient example 1](docs/img/restclient_cat_example_delete.png?raw=true)
 
-* `EitherRestErrorProcessor` returns values of type `EitherStrict<RestBusinessError, TResult>`. It checks if the response has an unsuccessful status code (one different than *2XX*). If it does, then it returns a `RestBusinessError`; if it doesn't, then it delegates the processing to the rest of the pipeline so it returns a `TResult` value.
+In the case of `DeleteCat`, we also need to pass `Unit.Default` to indicate that we don't want anything passed in the request body. If we needed to call DELETE passing a JSON object in the request body, we'd replace `Unit.Default` with the object to serialize.
+
+## Managing 404 status codes
+
+For a more complex example, imagine "GET http://api.catsforall.com/cats/myCatName" returns a _404 Not Found_ status code if the cat doesn't exist. With our current API, in that scenario we throw an exception. However, what if we want to handle that case programatically? We can do that in the following way:
+
+![restclient example 1](docs/img/restclient_cat_example_get_option.png?raw=true)
```

Comment:
let's put first the OptionAsNullProcessor, so can explain the usage of processors first, before digging into Monad